src/main/target/common_defaults_post.h

```diff
@@ -671,3 +671,12 @@
 #define DSHOT_BITBANGED_TIMER_DEFAULT DSHOT_BITBANGED_TIMER_AUTO
 #endif
 #endif // USE_DSHOT_BITBANG
+
+#define DMARXCAT(s) DMARCAT(s)
+#define DMARCAT(s) DMAR_##s
+#define DMAR_true  1
+#define DMAR_false 1
+
+#if DMARXCAT(ENABLE_DSHOT_DMAR) == 1 || DMARXCAT(ENABLE_DSHOT_DMAR) == 1
+#error "Use DSHOT_DMAR_ON or DSHOT_DMAR_OFF instead of boolean values for ENABLE_DSHOT_DMAR"
+#endif
```

Comment:
Not sure it's worth sticking this into here and not into `src/main/target/common_deprecated_post.h` where the rest of these warnings is - in general the problem is with target maintainers creating / updating targets based on deprecated templates, and not so much with defaults being changed.
On a somewhat related note, I think it can't hurt if we move evaluation of `src/main/target/common_deprecated_post.h` to after `src/main/target/common_defaults_post.h` and `src/main/target/common_post.h` in a separate pull request.

Reply:
Ok, sure. I put it into common_deprecated_post.h now. I just mechanically wanted to have it after any other defines of `ENABLE_DSHOT_DMAR` previously.